cobra/flux_analysis/sampling.py

```diff
@@ -84,14 +98,37 @@ class HRSampler(object):
     """
 
     def __init__(self, model, thinning, seed=None):
-        self.model = model
+        # This currently has to be done to reset the solver basis which is
+        # required to get deterministic warmup point generation
+        # (in turn required for a working `seed` arg)
+        if model.solver.is_integer:
+            raise TypeError("sampling does not work with integer problems :(")
+        self.model = model.copy()
         self.thinning = thinning
         self.n_samples = 0
-        self.S = create_stoichiometric_matrix(model, array_type='dense')
+        prob = constraint_matrices(model)
+        self.S = prob.equalities
```

Comment:
I think it would help readability to avoid the single letter attributes (S, NS, M, b etc, also BTOL, FTOL) in favor of complete names (and variables lower-case)